packages/component-base/src/iron-list-core.js

```diff
@@ -635,16 +638,21 @@ export const ironList = {
    * @param {boolean=} forceUpdate If true, updates the height no matter what.
    */
   _updateScrollerSize(forceUpdate) {
-    this._estScrollHeight =
+    const estScrollHeight =
       this._physicalBottom +
       Math.max(this._virtualCount - this._physicalCount - this._virtualStart, 0) * this._physicalAverage;
 
-    forceUpdate ||= this._scrollHeight === 0;
-    forceUpdate ||= this._scrollPosition >= this._estScrollHeight - this._physicalSize;
```

Comment:
Moved both lines to the `if` condition below as it's the only place where `forceUpdate` is used.